packages/coinstac-api-server/tests/resolvers.test.js

```diff
@@ -0,0 +1,1027 @@
+    t.is(err.message, 'error');
+  }
+
+  crypto.pbkdf2.restore();
+
+  /* verifyPassword */
+  const isValid1 = await helperFunctions.verifyPassword('password', passwordHash);
+  t.true(isValid1);
+
+  const isValid2 = await helperFunctions.verifyPassword('password');
+  t.false(isValid2);
+
+  sinon.stub(crypto, 'pbkdf2').yields(new Error('error'), null);
+
+  try {
+    await helperFunctions.verifyPassword('password', passwordHash);
+  } catch (error) {
+    t.is(error.message, 'error');
+  }
+
+  crypto.pbkdf2.restore();
+});
+
+test('validateToken and validateUser', async (t) => {
+  const users = await Query.fetchAllUsers();
+  const decoded1 = { id: users[0].id };
+
+  helperFunctions.validateToken(decoded1, null, (_, valid, user) => {
+    t.true(valid);
+    t.is(user.id, decoded1.id);
+  });
+
+  const decoded2 = { id: 'invalid-user' };
+
+  helperFunctions.validateToken(decoded2, null, (_, valid, user) => {
+    t.false(valid);
+    t.falsy(user);
+  });
+
+  const req1 = { payload: { username: 'test1', password: 'password' } };
+
+  await helperFunctions.validateUser(req1, (user) => {
+    t.is(user.username, req1.payload.username);
+  });
+
+  const req2 = { payload: { username: 'invalid-user', password: 'password' } };
+
+  await helperFunctions.validateUser(req2, (error) => {
+    t.is(error.output.payload.statusCode, 401);
+  });
+
+  const req3 = { payload: { username: 'test1', password: 'password1' } };
+
+  await helperFunctions.validateUser(req3, (error) => {
+    t.is(error.output.payload.statusCode, 401);
+  });
+});
+
+test('validateEmail', async (t) => {
+  let req = {
+    payload: { email: 'test@mrn.org' },
+  };
+
+  await helperFunctions.validateEmail(req, (exists) => {
+    t.true(exists);
+  });
+
+  req = {
+    payload: { email: 'email@mrn.org' },
+  };
+
+  await helperFunctions.validateEmail(req, (error) => {
+    t.is(getMessageFromError(error), INVALID_EMAIL);
+  });
+});
+
+test('validateUniqueEmail', async (t) => {
+  const req = { payload: { email: 'test@mrn.org' } };
+
+  const isValid = await helperFunctions.validateUniqueEmail(req);
+
+  t.false(isValid);
+});
+
+test('validateUniqueUsername', async (t) => {
+  const req = { payload: { username: 'valid-user' } };
+
+  const isUnique = await helperFunctions.validateUniqueUsername(req);
+
+  t.true(isUnique);
+});
+
+test('validateUniqueUser', async (t) => {
+  const req1 = { payload: { username: 'test1', email: 'test@mrn.org' } };
+
+  await helperFunctions.validateUniqueUser(req1, (error) => {
+    t.is(error.output.payload.message, 'Username taken');
+  });
+
+  const req2 = { payload: { username: 'newuser', email: 'test@mrn.org' } };
+
+  await helperFunctions.validateUniqueUser(req2, (error) => {
+    t.is(error.output.payload.message, 'Email taken');
+  });
+
+  const req3 = { payload: { username: 'newuser', email: 'newuser@mrn.org' } };
+  const res = sinon.spy();
+
+  await helperFunctions.validateUniqueUser(req3, res);
+  t.true(res.called);
```

Comment:
this is just saying this callback was called right? Could it have been called with an error? Why not just check if the error obj is false?